src/haddock/libs/librestraints.py

```diff
@@ -0,0 +1,627 @@
+        resid = int(struct.residueNumber(idx))
+        chain = struct.chainLabel(idx)
+        at_uid = (chain, resname, resid, atname)
+        res_uid = (chain, resname, resid)
+
+        asa = result.atomArea(idx)
+        asa_data[at_uid] = asa
+        # add asa to residue
+        rsa_data[res_uid] = rsa_data.get(res_uid, 0) + asa
+
+        if atname in ('C', 'N', 'O'):
+            rel_main_chain[res_uid] = rel_main_chain.get(res_uid, 0) + asa
+        else:
+            rel_side_chain[res_uid] = rel_side_chain.get(res_uid, 0) + asa
+
+    # convert total asa ro relative asa
+    rsa_data.update((res_uid, asa / _rsa[res_uid[1]]) for res_uid, asa in rsa_data.items())
+    rel_main_chain.update((res_uid, asa / _rsa_bb[res_uid[1]] * 100) for res_uid, asa in rel_main_chain.items())
+    rel_side_chain.update((res_uid, asa / _rsa_sc[res_uid[1]] * 100) for res_uid, asa in rel_side_chain.items())
+
+    # We format to fit the pipeline
+    resid_access = {}
+    for res_uid, access in rel_main_chain.items():
+        resid_access[res_uid[2]] = {'side_chain_rel': rel_side_chain.get(res_uid), 'main_chain_rel': access}
+    surface_resids = [r for r, v in resid_access.items() if v['side_chain_rel'] >= cutoff or
+                      v['main_chain_rel'] >= cutoff]
+    return surface_resids
+
+
+def parse_actpass_file(actpass_file):
+    """Parse actpass file
+    
+    Parameters
+    ----------
+    actpass_file: str or Path
+        path to actpass_file
+
+    Returns
+    -------
+    active: list
+        list of active residues
+    passive: list
+        list of passive residues
+    """
+    
+    if Path(actpass_file).exists() is False:
+        raise Exception(f"actpass file {actpass_file} does not exist.")
+
+    lines = open(actpass_file, "r").readlines()
+    nlines = len(lines)
+    if nlines != 2:
+        raise Exception(f"actpass file {actpass_file} does not have two lines (counted {nlines})")
+    active, passive = [[int(x) for x in line.split()] for line in lines]
+    return active, passive
+
+
+def active_passive_to_ambig(active1, passive1, active2, passive2, segid1='A', segid2='B'):
+    """Convert active and passive residues to Ambiguous Interaction Restraints
+
+    Parameters
+    ----------
+    active1 : list
+        List of active residue numbers of the first segid
+
+    passive1 : list
+        List of passive residue numbers of the first segid
+
+    passive2 : list
+        List of passive residue numbers of the second segid
+
+    active2 : list
+        List of active residue numbers of the second segid
+    
+    active2 : list
+        List of passive residue numbers of the second segid
+
+    segid1 : string
+        Segid to use for the first model
+
+    segid2 : string
+        Segid to use for the second model
+
+    """
+
+    all1 = active1 + passive1
+    all2 = active2 + passive2
+
+    for resi1 in active1:
+        print('assign (resi {:d} and segid {:s})'.format(resi1, segid1))
```

Comment:
Maybe a personal preference, but I'm not a big fan of these prints and would use `sys.stdout.write` instead but its a python nitpick, it's good as it is.